common/cpp/src/google_smart_card_common/value_nacl_pp_var_conversion_unittest.cc

```diff
@@ -0,0 +1,142 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <google_smart_card_common/value_nacl_pp_var_conversion.h>
+
+#include <stdint.h>
+
+#include <cstring>
+#include <limits>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <gtest/gtest.h>
+#include <ppapi/cpp/var.h>
+#include <ppapi/cpp/var_array.h>
+#include <ppapi/cpp/var_array_buffer.h>
+#include <ppapi/cpp/var_dictionary.h>
+
+#include <google_smart_card_common/optional.h>
+#include <google_smart_card_common/unique_ptr_utils.h>
+#include <google_smart_card_common/value.h>
+
+namespace google_smart_card {
+
+TEST(ValueNaclPpVarConversion, NullValue) {
+  EXPECT_TRUE(ConvertValueToPpVar(Value()).is_null());
+}
+
+TEST(ValueNaclPpVarConversion, BooleanValue) {
+  const bool kBoolean = 123;
+  const pp::Var var = ConvertValueToPpVar(Value(kBoolean));
+  ASSERT_TRUE(var.is_bool());
+  EXPECT_EQ(var.AsBool(), kBoolean);
+}
+
+TEST(ValueNaclPpVarConversion, IntegerValue) {
+  const int kInteger = 123;
+  const pp::Var var = ConvertValueToPpVar(Value(kInteger));
+  ASSERT_TRUE(var.is_int());
+  EXPECT_EQ(var.AsInt(), kInteger);
+}
+
+TEST(ValueNaclPpVarConversion, Integer64BitMaxValue) {
+  const int64_t k64BitMax = std::numeric_limits<int64_t>::max();
+  const pp::Var var = ConvertValueToPpVar(Value(k64BitMax));
+  ASSERT_TRUE(var.is_double());
+  EXPECT_DOUBLE_EQ(var.AsDouble(), k64BitMax);
+}
+
+TEST(ValueNaclPpVarConversion, Integer64BitMinValue) {
+  const int64_t k64BitMin = std::numeric_limits<int64_t>::min();
+  const pp::Var var = ConvertValueToPpVar(Value(k64BitMin));
+  ASSERT_TRUE(var.is_double());
+  EXPECT_DOUBLE_EQ(var.AsDouble(), k64BitMin);
+}
+
+TEST(ValueNaclPpVarConversion, FloatValue) {
+  const double kFloat = 123.456;
+  const pp::Var var = ConvertValueToPpVar(Value(kFloat));
+  ASSERT_TRUE(var.is_double());
+  EXPECT_DOUBLE_EQ(var.AsDouble(), kFloat);
+}
+
+TEST(ValueNaclPpVarConversion, StringValue) {
+  const char kString[] = "foo";
+  const pp::Var var = ConvertValueToPpVar(Value(kString));
+  ASSERT_TRUE(var.is_string());
+  EXPECT_EQ(var.AsString(), kString);
+}
+
+TEST(ValueNaclPpVarConversion, BinaryValue) {
+  const std::vector<uint8_t> kBinary = {1, 2, 3};
+  const pp::Var var = ConvertValueToPpVar(Value(kBinary));
+  ASSERT_TRUE(var.is_array_buffer());
+  pp::VarArrayBuffer var_array_buffer(var);
+  EXPECT_EQ(var_array_buffer.ByteLength(), kBinary.size());
+  const void* const buffer_contents = var_array_buffer.Map();
```

Comment:
should we always call Unmap() after Map()?

Reply:
Not strictly required (as this will be done when the variable goes out of scope), but let's do that for consistency.